systemd/esm-cache.service

```diff
@@ -13,3 +13,13 @@ Description=Update the local ESM caches
 [Service]
 Type=oneshot
 ExecStart=/usr/bin/python3 /usr/lib/ubuntu-advantage/esm_cache.py
+AppArmorProfile=-ubuntu_pro_esm_cache
+CapabilityBoundingSet=~CAP_SYS_ADMIN
+CapabilityBoundingSet=~CAP_NET_ADMIN
+CapabilityBoundingSet=~CAP_NET_BIND_SERVICE
+CapabilityBoundingSet=~CAP_SYS_PTRACE
+CapabilityBoundingSet=~CAP_NET_RAW
+PrivateTmp=true
+RestrictAddressFamilies=~AF_NETLINK
+RestrictAddressFamilies=~AF_PACKET
```

Comment:
Is this a best effort hardening approach?

I wonder how the fact that some of our architectures have no support to RestrictAddressFamilies interferes here (https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#RestrictAddressFamilies=)

> Note that this option has no effect on 32-bit x86, s390, s390x, mips, mips-le, ppc, ppc-le, ppc64, ppc64-le and is ignored (but works correctly on other ABIs, including x86-64).

Cc @panlinux

Reply:
Not sure what you mean by best efford specifically regarding address families. I removed the ones that were obvious to me that esm-cache.service would not need. Regarding the lack of support in other architectures, I didn't know that, but since it doesn't seem to produce a failure, I think we can keep them instead of introducing a rendering mechanism for esm-cache.service that would only add those lines in the architectures where it makes sense.

Reply:
I see. I was just wondering if we were aware of the per-architecture differences in this case.

Reply:
I wasn't, so thanks for that. It doesn't seem to affect things, and another ubuntu-advantage-tools SRU was already done that had these same restrictions for apt-news.service.